Source/WebCore/dom/ComposedTreeIterator.h

```diff
@@ -136,42 +142,45 @@ inline ComposedTreeIterator& ComposedTreeIterator::traversePreviousSibling()
     return *this;
 }
 
-inline unsigned ComposedTreeIterator::depth() const
+template <size_t ContextInlineCapacity>
+inline unsigned ComposedTreeIterator<ContextInlineCapacity>::depth() const
 {
     unsigned depth = 0;
     for (auto& context : m_contextStack)
         depth += context.iterator.depth();
     return depth;
 }
 
+template <size_t ContextInlineCapacity = defaultInlineCapacity>
 class ComposedTreeDescendantAdapter {
 public:
     ComposedTreeDescendantAdapter(ContainerNode& parent)
         : m_parent(parent)
     { }
 
-    ComposedTreeIterator begin() { return ComposedTreeIterator(m_parent, ComposedTreeIterator::FirstChild); }
-    ComposedTreeIterator end() { return { }; }
-    ComposedTreeIterator at(const Node& child) { return ComposedTreeIterator(m_parent, const_cast<Node&>(child)); }
-    
+    ComposedTreeIterator<ContextInlineCapacity> begin() { return ComposedTreeIterator<ContextInlineCapacity>(m_parent, ComposedTreeIterator<ContextInlineCapacity>::FirstChild); }
+    ComposedTreeIterator<ContextInlineCapacity> end() { return { }; }
+    ComposedTreeIterator<ContextInlineCapacity> at(const Node& child) { return ComposedTreeIterator(m_parent, const_cast<Node&>(child)); }
+
 private:
     CheckedRef<ContainerNode> m_parent;
 };
 
+template <size_t ContextInlineCapacity = defaultInlineCapacity>
 class ComposedTreeChildAdapter {
```

Comment:
child adapter could probably have less or no inline capacity by default but I guess this is safest for this patch